src/lib/desmoHub-module.ts

```diff
@@ -1,325 +1,374 @@
+      );
+    }
+
+    const tx = await this.contract.enableTDD();
+    this.TRANSACTION_SENT.next({
+      invokedOperation: OperationType.enableTDD,
+      hash: tx.hash,
+      sent: new Date(Date.now()),
+    });
+  }
+/**
+ * This method is used to call the homonym function on the smart contract to get a new Request id.
+ * It produce an event when the transaction is sent.
+ * It is possible to get the result of the transaction by subscribing to the requestID$ observable, after having activated the listeners.
+ */
+  public async getNewRequestID(): Promise<void> {
+    if (!this.isConnected) {
+      throw new Error(
+        'This method requires the wallet signer to be already signed-in!',
+      );
+    }
+
+    const tx = await this.contract.getNewRequestID();
+    this.TRANSACTION_SENT.next({
+      invokedOperation: OperationType.getNewRequestID,
+      hash: tx.hash,
+      sent: new Date(Date.now()),
+    });
+  }
+/**
+ * 
+ * @returns the length of the TDD storage.
+ */
+  public async getTDDStorageLength(): Promise<ethers.BigNumber> {
+    return await this.contract.getTDDStorageLength();
+  }
+/**
+ * Get the scores relative to a request id.
+ * 
+ * @param requestID 
+ * @returns 
+ */
+  public async getScoresByRequestID(
+    requestID: ethers.Bytes,
+  ): Promise<ethers.Bytes> {
+    return await this.contract.getScoresByRequestID(requestID);
+  }
+/**
+ * Get TDD by request id.
+ * 
+ * @param requestID 
+ * @returns 
+ */
+  public async getTDDByRequestID(requestID: ethers.Bytes): Promise<string[]> {
+    return await this.contract.getTDDByRequestID(requestID);
+  }
+
+  /**
+   * 
+   * @returns the TDD of the current user.
+   */
+  public async getTDD(): Promise<ITDD> {
+    const { url, owner, disabled, score }: ITDD = await this.contract.getTDD();
+    return { url, owner, disabled, score } as ITDD;
+  }
+
+  /**
+   * @param start the initial TDD index to retrieve
+   * @param stop the index at which to stop retrieving TDDs (if not specified, the full length of the TDD storage inside the contract)
+   * @returns an array containing the requested TDD descriptions.
+   */
+  public async getTDDList(start = 0, stop?: number): Promise<ITDD[]> {
+    const tddStorageLength = (await this.getTDDStorageLength()).toNumber();
+
+    if (tddStorageLength < 1) {
+      throw new Error('No TDDs are currently registered in the contract.');
+    }
+
+    // Make sure that `start` and `stop` are integer numbers:
+    start = Math.floor(start ?? 0);
+    stop = Math.floor(stop ?? tddStorageLength);
+
+    // Apply limits to `start` and `stop` indexes:
+    start = Math.max(start, 0);
+    stop = Math.min(stop, tddStorageLength);
+
+    if (start >= tddStorageLength) {
+      throw new Error(`Start index must be lower than the TDD storage length (${tddStorageLength}).`);
+    } else if (start > stop) {
+      throw new Error(
+        `Start index (${start}) is greater than stop index (${stop}).`,
+      );
+    }
```

Comment:
No need to use `else if` here, do a regular `if` is cleaner.

Reply:
Done.